lib/data_table.rb

```diff
@@ -977,6 +943,60 @@ def parameter_binding_character
         is_connection_sql_server? ? /[\$@].+/ : /\$.+/
       end
 
+      def adjust_user_visibility(columns, options)
+        return unless columns.present? && options.dig(:columns_layout, 'columns')
+
+        columns.each do |column|
+          user_column = find_user_column(column.name, options)
+          column.options.reverse_merge!(:user_visible => user_column['visible'], :user_visible_index => user_column['visibleIndex'])
+        end
+
+        columns.sort_by! { |c| c.options[:user_visible_index] }
+      end
+
+      def find_user_column(column_name, options)
+        options[:columns_layout]['columns'].detect do |c|
+          split_layout_name = c['dataField'].split('.')
+          name = split_layout_name.length > 2 ? split_layout_name.last(2).map(&:to_sym).to_s : split_layout_name.last.to_s
+          column_name.to_s == name
+        end || { 'visible' => false }
+      end
+
+      def generate_csv_output_stream(csv_output_stream, columns)
```

Comment:
@nbrien I guess you were talking about this method?

Either the gem needs to implement the SQL and PG version and check the connection type for tyhe implementation?
Or it should be an abstracto method and each applicatyion implements their own streaming (This is less re-usable though from multiple projects).

@WillemOpperman what is your input here?

Reply:
I agree. This will not work for RC.

This code should be in robusta/rc-ui.